### PR TITLE
feature: Add support for `immediate` flag when using Kinesis analytics providers

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change adds support for the `immediate` flag when using `AWSKinesisProvider` or `AWSKinesisFirehouseProvider` to align with our published API contract. Setting the `immediate` flag on a record will trigger the providers to immediately send the record to the appropriate Kinesis client without trying to buffer it. The motivations for this change were to align the API with Pinpoint client & help a customer who was having trouble emitted Kinesis events from a RN application while in the background (due to RN pausing timers, such as the one used to flush the event buffer).

#### Issue #, if available

SIM Issue

#### Description of how you validated changes
Local testing, clean integration test run. Verified that `immediate` records were persisted in the Kinesis data stream in a React Native application from both the foreground and background (see attached image).

Screen Shot 2022-12-16 at 9 00 32 AM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/analytics/src/Providers/AWSKinesisProvider.ts [hunk 1]
```diff
@@ -89,7 +89,13 @@ export class AWSKinesisProvider implements AnalyticsProvider {
 
 		Object.assign(params, { config: this._config, credentials });
 
-		return this._putToBuffer(params);
+		if (params.event?.immediate) {
+			this._sendEvents([params]);
+
+			return Promise.resolve(true);
```
Comment (jimblanc) [inline packages/analytics/src/Providers/AWSKinesisProvider.ts:95]: This matches the behavior of `this._putToBuffer`

**Review by AllanZhengYP — APPROVED** [review]
It looks good to me. Is there any documentation update required?

**Comment by jimblanc** [comment]
> It looks good to me. Is there any documentation update required?

Yep there was a 1 line change required, working on it now.

**Review by elorzafe — APPROVED** [review]
Looks good to me

Thanks @jimblanc 🔥

**Pull request merged**